gui/src/ts/components/kg/node/KgNodeTable.tsx

```diff
@@ -25,66 +15,158 @@ const showListAsColumn = (list: string[]) =>
+  onChangeRowsPerPage,
+}) => {
+  // https://github.com/gregnb/mui-datatables/issues/756
+  // Since the MUIDataTable has its own state, it ignores passed in values
+  // for page and rowsPerPage
+  // Workaround by checking the table state via ref and updating
+  // with internal api
+  const tableCallbackRef = (table: any) => {
+    if (!table) {
+      return;
+    }
+
+    const {page: tablePage, rowsPerPage: tableRowsPerPage} = table.state;
+
+    if (tableRowsPerPage !== rowsPerPage) {
+      table.changeRowsPerPage(rowsPerPage);
+    }
+
+    if (tablePage !== page) {
+      table.changePage(page);
+    }
+  };
+
   return (
-    <TableContainer component={Paper}>
-      <Table data-cy="matchingNodesTable">
-        <TableHead>
-          <TableRow>
-            <TableCell></TableCell>
-            <TableCell>Label</TableCell>
-            <TableCell>Aliases</TableCell>
-            <TableCell>Datasource</TableCell>
-            <TableCell>Other</TableCell>
-            <TableCell>Pos</TableCell>
-          </TableRow>
-        </TableHead>
-        <TableBody>
-          {nodes.map((node, index) => (
-            <TableRow key={node.id}>
-              <TableCell>{page * rowsPerPage + index + 1}</TableCell>
-              <TableCell>
-                <KgNodeLink node={node} />
-              </TableCell>
-              <TableCell>
-                {node.aliases && showListAsColumn(node.aliases)}
-              </TableCell>
-              <TableCell>
-                {node.datasource &&
-                  node.datasource.split(",").map((datasource) => (
-                    <React.Fragment key={datasource}>
-                      <KgDatasourceLink datasource={datasource} />
-                      <br />
-                    </React.Fragment>
-                  ))}
-              </TableCell>
-              <TableCell>{node.other}</TableCell>
-              <TableCell>
-                {node.pos && showListAsColumn(node.pos.split(","))}
-              </TableCell>
-            </TableRow>
-          ))}
-        </TableBody>
-        <TableFooter>
-          <TablePagination
-            rowsPerPage={rowsPerPage}
-            count={count}
-            page={page}
-            onChangePage={(_, newPage) => onChangePage(newPage)}
-            onChangeRowsPerPage={(event) =>
-              onChangeRowsPerPage(+event.target.value)
-            }
-            ActionsComponent={TablePaginationActions}
-          />
-        </TableFooter>
-      </Table>
-    </TableContainer>
+    <div data-cy="matchingNodesTable">
+      <MUIDataTable
+        // Is a @material-ui property
+        // but missing in types
+        // @ts-ignore
+        innerRef={tableCallbackRef}
+        title={
+          <Typography variant="h6" data-cy="title">
+            {title}
+          </Typography>
+        }
+        data={nodes}
+        columns={columns}
+        options={{
+          count,
+          serverSide: true,
+          sort: false,
```

Comment:
We'll need to implement sorting on the server side. I filed a new issue for that.

Reply:
Sounds good